test/e2e/network/dns.go

```diff
@@ -67,12 +66,39 @@ var _ = SIGDescribe("DNS", func() {
 		validateDNSResults(f, pod, append(wheezyFileNames, jessieFileNames...))
 	})
 
+	It("should resolve DNS of partial qualified names for the cluster ", func() {
+		// All the names we need to be able to resolve.
+		// TODO: Spin up a separate test service and test that dns works for that service.
+		namesToResolve := []string{
+			"kubernetes.default",
+			"kubernetes.default.svc",
+			"kubernetes.default.svc.cluster",
+		}
+		// Added due to #8512. This is critical for GCE and GKE deployments.
+		if framework.ProviderIs("gce", "gke") {
+			namesToResolve = append(namesToResolve, "google.com")
+			namesToResolve = append(namesToResolve, "metadata")
+		}
+		hostFQDN := fmt.Sprintf("%s.%s.%s.svc.cluster.local", dnsTestPodHostName, dnsTestServiceName, f.Namespace.Name)
```

Comment:
if we _are_ ok splitting up DNS into conformance/extra test cases, can we refactor to de-dupe the actual test mechanics?

Reply:
Sure